lib/OpenQA/CacheService/Task/Sync.pm

```diff
@@ -39,10 +41,17 @@ sub _cache_tests {
     my @cmd = (qw(rsync -avHP --timeout), RSYNC_TIMEOUT, "$from/", qw(--delete), "$to/tests/");
     my $cmd = join ' ', @cmd;
     $ctx->info("Calling: $cmd");
-    my $output = `@cmd`;
-    my $status = $? >> 8;
+    my $status;
+    my $full_output = "";
+    for my $retry (1 .. RSYNC_RETRIES) {
+        my $output = `@cmd`;
+        $status = $? >> 8;
+        $full_output .= "Try $retry:\n" . $output . "\n";
+        last unless $status;
+        sleep RSYNC_RETRY_PERIOD; # uncoverable statement
```

Comment:
Ok, so this line isn't executed anyways by tests. How do you come to the conclusion that my suggestion to set `OPENQA_RSYNC_RETRY_PERIOD` to 0 isn't working then? Of course if the line isn't executed anyways we don't have to worry much about slowing down tests although it would likely be nice to actually have a test which runs all of the code.

Reply:
https://github.com/os-autoinst/openQA/pull/4464/files#diff-9a2b1ade12881dfa535761e60a94118eabd51ecdad9a003037400c6cad1df61dR114-R117 this should execute whole for loop include `sleep` statement ...

Reply:
Then we shouldn't need to mark the line as uncoverable statement (unless it is running in a way which doesn't allow us to track its coverage, e.g. in a forked process).

Reply:
@kraih how `sleep` behave in minion job?

Reply:
> Then we shouldn't need to mark the line as uncoverable statement (unless it is running in a way which doesn't allow us to track its coverage, e.g. in a forked process).

so question is why codecov consider this line not running.. whed I added debug lines before and after they show correctly in test run

Reply:
Oh, when it is within a Minion job that's a forked process and therefore the coverage collection doesn't work. We actually have the utility function `perform_minion_jobs` to avoid forking which might be applicable here as well.

Reply:
What @Martchus said. It's a bit tricky, since depending on test strategy Minion jobs might run without forking.

Reply:
no idea how to use `perform_minion_jobs` in this case

Reply:
I'm trying to add `perform_minion_jobs`, but this will take a while.